docs/content.zh/docs/dev/datastream/fault-tolerance/state_v2.md

```diff
@@ -0,0 +1,601 @@
+that you can reference them), the type of the values that the state holds, and possibly
+a user-specified function, such as a `ReduceFunction`. Depending on what type of state you
+want to retrieve, you create either a `ValueStateDescriptor`, a `ListStateDescriptor`, 
+an `AggregatingStateDescriptor`, a `ReducingStateDescriptor`, or a `MapStateDescriptor`.
+To differentiate between the previous state APIs, you should use the `StateDescriptor`s under the
+`org.apache.flink.api.common.state.v2` package (note the **v2**).
+
+State is accessed using the `RuntimeContext`, so it is only possible in *rich functions*.
+Please see [here]({{< ref "docs/dev/datastream/user_defined_functions" >}}#rich-functions) for
+information about that, but we will also see an example shortly. The `RuntimeContext` that
+is available in a `RichFunction` has these methods for accessing state:
+
+* `ValueState<T> getState(ValueStateDescriptor<T>)`
+* `ReducingState<T> getReducingState(ReducingStateDescriptor<T>)`
+* `ListState<T> getListState(ListStateDescriptor<T>)`
+* `AggregatingState<IN, OUT> getAggregatingState(AggregatingStateDescriptor<IN, ACC, OUT>)`
+* `MapState<UK, UV> getMapState(MapStateDescriptor<UK, UV>)`
+
+This is an example `FlatMapFunction` that shows how all of the parts fit together:
+
+{{< tabs "348fd48c-fa36-4d27-9e53-9bffb74d8a87" >}}
+{{< tab "Java" >}}
+```java
+public class CountWindowAverage extends RichFlatMapFunction<Tuple2<Long, Long>, Tuple2<Long, Long>> {
+
+    /**
+     * The ValueState handle. The first field is the count, the second field a running sum.
+     */
+    private transient ValueState<Tuple2<Long, Long>> sum;
+
+    @Override
+    public void flatMap(Tuple2<Long, Long> input, Collector<Tuple2<Long, Long>> out) throws Exception {
+
+        // access the state value
+        sum.asyncValue().thenApply(currentSum -> {
+            // if it hasn't been used before, it will be null
+            Tuple2<Long, Long> current = currentSum == null ? Tuple2.of(0L, 0L) : currentSum;
+
+            // update the count
+            current.f0 += 1;
+
+            // add the second field of the input value
+            current.f1 += input.f1;
+
+            return current;
+        }).thenAccept(r -> {
+            // if the count reaches 2, emit the average and clear the state
+            if (r.f0 >= 2) {
+                out.collect(Tuple2.of(input.f0, r.f1 / r.f0));
+                sum.asyncClear();
+            } else {
+                sum.asyncUpdate(r);
+            }
+        });
+    }
+
+    @Override
+    public void open(OpenContext ctx) {
+        ValueStateDescriptor<Tuple2<Long, Long>> descriptor =
+                new ValueStateDescriptor<>(
+                        "average", // the state name
+                        TypeInformation.of(new TypeHint<Tuple2<Long, Long>>() {})); // type information
+        sum = getRuntimeContext().getState(descriptor);
```

Comment:
Should we convert `getRuntimeContext()` to `StreamingRuntimeContext`?

Reply:
I thought this is a bug. I opened #26120 to fix this.